TESTING.md

```diff
@@ -0,0 +1,57 @@
+# Testing our components with Jest
+
+“Jest is a JavaScript testing framework designed to ensure correctness of any JavaScript codebase. It allows you to write tests with an approachable, familiar and feature-rich API that gives you results quickly.” 
+
+These instructions have been written with JavaScript in mind. Jest can also be used with TypeScript by substituting `.js` test file extensions with `.ts`.
+
+## Installing Jest
+
+Install Jest with `npm` or `yarn`.
+
+```shell
+npm install --save-dev jest
+
+yarn add --dev jest
+```
+
+Create a test file using the extension: `.test.js` and add some tests e.g.
+
+```js
+<>
+```
+
+Add the following section to your package.json:
+
+```shell
+{
+  "scripts": {
+    "test": "jest"
+  }
+}
+```
+
+Finally, run your tests using `npm test` or `yarn test`.
+
+More information on setting up Jest can be found on [Jest’s Getting Started page]( https://jestjs.io/docs/getting-started).
+
+
+## What is the Shadow DOM?
+
+A key part of our packages is the shadow DOM. With the shadow DOM, we can encapsulate each component’s markup structure, styling and behaviour so different parts don’t clash.
+
+According to Mozilla, the “shadow DOM allows hidden DOM trees to be attached to elements in the regular DOM tree — this shadow DOM tree starts with a shadow root, underneath which you can attach any element, in the same way as the normal DOM.”
+
+### Testing the Shadow DOM
+
+Find elements in the shadow DOM using `shadowRoot`. The shadow root is the root node of the shadow DOM tree.
+
+#### Example
+```js
+// To find all navigation items in a top nav
+document.querySelector(“ic-top-navigation).querySelectorAll(“ic-navigation-item”)
```

Comment:
Do we need to say something as to why navigation items are not in the shadow DOM themselves, i.e. because they are slotted elements